contracts/Vault.vy

```diff
@@ -1774,6 +1767,37 @@ def report(gain: uint256, loss: uint256, _debtPayment: uint256) -> uint256:
 
     # Update reporting time
     self.strategies[msg.sender].lastReport = block.timestamp
+
+    # maintains longer (fairer) harvest periods on close timed harvests
+    # NOTE: correctly adjust time delta to avoid reducing locked-until time
+    #       all following examples have previousHarvestTimeDelta = 10 set at h2 and used on h3
+    #       if new time delta reduces previous locked-until, keep locked-until and adjust remaining time
+    #       h1 = t0, h2 = t10 and h3 = t13 =>
+    #           currentTimeDelta = 3, (new)previousHarvestTimeDelta = 7 (10-3), locked until t20
+    #       h1 = t0, h2 = t10 and h3 = t14 =>
+    #           currentTimeDelta = 4, (new)previousHarvestTimeDelta = 6 (10-4), locked until t20
+    #       on 2nd example: h2 is getting carried into h3 (minus time delta 4) since it was previously trying to reach t20.
+    #       so it continues to spread the lock up to that point, and thus avoids reducing the previous distribution time.
+    #
+    #       if locked-until is unchanged, to avoid extra storage read and subtraction cost [behaves as examples below]
+    #       h1 = t0, h2 = t10 and h3 = t15 =>
+    #           currentTimeDelta = 5, (new)previousHarvestTimeDelta = 5 locked until t20
+    #
+    #       if next total time delta is higher than previous period remaining, locked-until will increase
+    #       h1 = t0, h2 = t10 and h3 = t16 =>
+    #           currentTimeDelta = 6, (new)previousHarvestTimeDelta = 6 locked until t22
+    #       h1 = t0, h2 = t10 and h3 = t17 =>
+    #           currentTimeDelta = 7, (new)previousHarvestTimeDelta = 7 locked until t24
+    #       
+    #       currentTimeDelta is the time delta between now and lastReport.
+    #       previousHarvestTimeDelta is the time delta between lastReport and the previous lastReport
+    #       previousHarvestTimeDelta is assigned the higher value between currentTimeDelta and (previousHarvestTimeDelta - currentTimeDelta)
+    currentTimeDelta: uint256 = block.timestamp - self.lastReport
+    if self.previousHarvestTimeDelta > currentTimeDelta * 2:
+        self.previousHarvestTimeDelta = self.previousHarvestTimeDelta - currentTimeDelta
+    else:
+        self.previousHarvestTimeDelta = currentTimeDelta
```

Comment:
One thing to take into consideration is that if a harvest period is "late" (aka Keep3r issues or something causing a prolonged period without reports), then when it becomes active again it will delay profit much longer than expected.

Not sure if this is a desireable quality, but if not we could "saturate" the timing here by placing a governance-configurable "maximum" on the delay (e.g. 24 hours)

Reply:
that should be the desired behavior, since if harvest get delayed, it potentially had a higher impact in PPS, and we'll be, by setting a max delay, creating an incentive for depositors to snipe those delayed-harvests to get unfair increase on their deposits for a lower period.

Reply:
Let's say the max delay is 24 hrs, and the profit is $50k because we haven't harvested in a week. That means the fade-in profit would be something like $50k/24hr = $2k/hr, total accross all depositors. I would argue that it is still doing it's job of discouraging sandwiches, but with improved UX of not having to wait another week for full profit to be dispersed (actually 2 whole weeks, wouldn't it?)

At the very least, we can set it to something large enough to be effectively "disabled" but use it as an extra switch in times of need. Just my 2 cents, we are losing a lever of control here (which is arguably a good thing at a high level)

Reply:
I'd like to go without the max delay.
in those rare cases where we do have really big delays, subsequent harvests will not push the rewards further back, but they'll increase the total profit rate. (if they were profitable harvests)
if this happens a user can jump in right before the harvest and get 7x rewards on their deposit over 24hs.
We want an incentive for users to stay for as long as possible. delaying profits does not affect APY, it makes it more stable and/or predictable.
It's also one less thing to worry about setting up correctly.
I think tho, that this needs further testing and experimentation to see what will happen on those edge cases. (also when quickly reporting a loss after a big delayed profitable harvest)